### ISSUE TITLE
Getting wrong type in `conformingTo protocols` in `ExtensionMacro`

### ISSUE THREAD

**Issue by soumyamahunt** [swiftlang/swift-syntax#2049]
### Description

When declaring an extension macro that provides extensions for multiple protocols, i.e. `Codable` macro providing extension for `Decodable` and `Encodable`, by default protocol names are joined together and provided as a single type in the `protocols` array. 

For each protocol conformance is provided, `protocols` array should have a separate type syntax item.

### Steps to Reproduce

Xcode: Version 15.0 beta 6 (15A5219j)
Swift syntax version: 509.0.0-swift-DEVELOPMENT-SNAPSHOT-2023-08-07-a

Create an extension macro implementation that should print the `protocols` array in build log:

```swift
public struct CodableMacro: ExtensionMacro {
    public static func expansion(
        of node: AttributeSyntax,
        attachedTo declaration: some DeclGroupSyntax,
        providingExtensionsOf type: some TypeSyntaxProtocol,
        conformingTo protocols: [TypeSyntax],
        in context: some MacroExpansionContext
    ) throws -> [ExtensionDeclSyntax] {
        precondition(protocols.count == 0, "\(protocols)")
        return []
    }
}
```

```swift
@attached(extension, conformances: Decodable, Encodable)
public macro Codable() = #externalMacro(module: "ExtensionMacroMacros", type: "CodableMacro")
```

And attach this macro to a type:

```swift
@Codable
struct SomeCodable {
    let val: String
}
```

Inspect the `protocols` array in build log:
> [IdentifierTypeSyntax
╰─name: identifier("DecodableEncodable")]

**Comment by ahoppen** [comment]
Tracked in Apple’s issue tracker as rdar://113668922

**Comment by soumyamahunt** [comment]
Thanks @ahoppen and @hborla for the quick fix, let me know once fix is available in latest snapshots. This is a major blocker for [MetaCodable](https://github.com/SwiftyLab/MetaCodable) migrating to extension macros.

**Comment by ahoppen** [comment]
The [August 12 main development snapshot](https://www.swift.org/download/#trunk-development-main) should contain the fix.

**Comment by soumyamahunt** [comment]
> The [August 12 main development snapshot](https://www.swift.org/download/#trunk-development-main) should contain the fix.

Thanks @ahoppen, with latest main snapshot this issue is fixed, [MetaCodable](https://github.com/SwiftyLab- /MetaCodable) is buildable now (although tests are failing due to #2031). Closing this issue.

**Comment by soumyamahunt** [comment]
@ahoppen getting this issue again with latest Xcode release and swift-syntax version:

Xcode: Version 15.0 (15A240d)
Swift syntax version: 509.0.0

> IdentifierTypeSyntax
╰─name: identifier("DecodableEncodable")

**Comment by ahoppen** [comment]
The fix hasn’t made it into Xcode 15.0, unfortunately. As this is fixed in the GitHub repo, the fix should be included in a future Xcode version.
